src/core/public/apm_system.test.ts

```diff
@@ -0,0 +1,176 @@
+      const apmSystem = new ApmSystem({ active: true, globalLabels: { alpha: 'one' } });
+      await apmSystem.setup();
+      expect(apm.addLabels).toHaveBeenCalledWith({ alpha: 'one' });
+    });
+
+    describe('http request normalization', () => {
+      let windowSpy: any;
+
+      beforeEach(() => {
+        windowSpy = jest.spyOn(global as any, 'window', 'get').mockImplementation(() => ({
+          location: {
+            protocol: 'http:',
+            hostname: 'mykibanadomain.com',
+            port: '5601',
+          },
+        }));
+      });
+
+      afterEach(() => {
+        windowSpy.mockRestore();
+      });
+
+      it('adds an observe function', async () => {
+        const apmSystem = new ApmSystem({ active: true });
+        await apmSystem.setup();
+        expect(apm.observe).toHaveBeenCalledWith('transaction:end', expect.any(Function));
+      });
+
+      /**
+       * Utility function to wrap functions that mutate their input but don't return the mutated value.
+       * Makes expects easier below.
+       */
+      const returnArg = <T>(func: (input: T) => any): ((input: T) => T) => {
+        return (input) => {
+          func(input);
+          return input;
+        };
+      };
+
+      it('removes the hostname, port, and protocol only when all match window.location', async () => {
+        const apmSystem = new ApmSystem({ active: true });
+        await apmSystem.setup();
+        const observer = apmMock.observe.mock.calls[0][1];
+        const wrappedObserver = returnArg(observer);
+
+        // Strips the hostname, protocol, and port from URLs that are on the same origin
+        expect(
+          wrappedObserver({
+            type: 'http-request',
+            name: 'GET http://mykibanadomain.com:5601/asdf/qwerty',
+          } as Transaction)
+        ).toEqual({ type: 'http-request', name: 'GET /asdf/qwerty' });
+
+        // Does not modify URLs that are not on the same origin
+        expect(
+          wrappedObserver({
+            type: 'http-request',
+            name: 'GET https://mykibanadomain.com:5601/asdf/qwerty',
+          } as Transaction)
+        ).toEqual({
+          type: 'http-request',
+          name: 'GET https://mykibanadomain.com:5601/asdf/qwerty',
+        });
+
+        expect(
+          wrappedObserver({
+            type: 'http-request',
+            name: 'GET http://mykibanadomain.com:9200/asdf/qwerty',
+          } as Transaction)
+        ).toEqual({
+          type: 'http-request',
+          name: 'GET http://mykibanadomain.com:9200/asdf/qwerty',
+        });
+
+        expect(
+          wrappedObserver({
+            type: 'http-request',
+            name: 'GET http://myotherdomain.com:5601/asdf/qwerty',
+          } as Transaction)
+        ).toEqual({
+          type: 'http-request',
+          name: 'GET http://myotherdomain.com:5601/asdf/qwerty',
+        });
+      });
+
+      it('strips the basePath', async () => {
```

Comment:
It always has been a little misleading that we are using the same terminology for server's basePath and request's basePath

Reply:
Yeah we have basePath and serverBasePath but that isn't very clear :)